src/dguweb/web/controllers/search_controller.ex

```diff
@@ -5,14 +5,12 @@ defmodule DGUWeb.SearchController do
     render conn, :index
   end
 
-
-  def search(conn, %{}), do: showall(conn)
   def search(conn, %{"q" => ""}), do: showall(conn)
-
   def search(conn, %{"q" => query}) do
     result = URI.encode(query) |> Repo.search
     render conn, :search, query: query, results: result.hits.hits, total: result.hits.total
   end
+  def search(conn, %{}), do: showall(conn)
```

Comment:
Nitpicking: the vertical spacing between function definitions is a bit inconsistent.